### DISCUSSION TITLE
Moving forward

### DISCUSSION THREAD

**Discussion by tteck** [tteck/Proxmox#4025] [Announcements]
The community needs to unite and make some important decisions about the path forward. Has a new fork become the go-to choice? Is there a new frontend that we’re rallying behind? This site will soon be archived, and time is limited. Let’s make these decisions together.

**Comment by zenodotus280** [comment]
I just read your update on the other discussion - many thanks for your great work!

Is there a fork of this project already? Which frontend are you referring to? I only access these scripts through GitHub directly. I hope there is someone who can bear the torch!

**Reply by bramsuurdje** [reply]
if you go to my github profile you will see a proxmox-helper-scripts repo. that is an alternative website that i have created. as for forks. to my understanding there and none as of right now

**Comment by XPGeek** [comment]
Only recently discovered this, nothing productive to add, but a big thank you from the Internet ether. May life treat you as well as you have treated us, godspeed.

**Comment by MickLesk** [comment]
How about transitioning to a community-driven project structure? Similar to the MagicMirror project, which began with one developer handling pull requests and evolved into a community-led effort, we could adopt a model where key contributors take on the responsibility of maintaining and developing the project together.

This approach would create a structured environment, ensuring that the project remains active and evolves collaboratively. With the original site set to be archived soon, a community-driven model could offer both sustainability and continuity.

I’d be willing to take on such a role, and I believe the other three active pull request contributors would be open to it as well. (@havardthom @remz1337 @nicedevil007 ....)

**Reply by lslvxy** [reply]
Recommended to be driven and maintained by the community. These scripts are very useful to me, providing a great deal of help.

**Reply by Mellowlynx** [reply]
I agree with this, as I'm still learning to code, I will probably not contribute scripts anytime soon, but happy to provide help and test/contribute where I can.

**Reply by CrazyWolf13** [reply]
We just need to keep security in mind, Imagine a bad actor getting access to this repo and adding malicious code, in probably some days a huuge amount of machines would be compromised.

**Reply by Zedris** [reply]
I think this is a sensible idea just need to be aware of the security aspect. Maybe the 3 most active contributors at the start take a maintainer role and maybe after move to the community version you mentioned.

**Reply by MickLesk** [reply]
Yep, there has to be a 2 or 3 eyes-check before any code is merged into the project.

**Reply by tteck** [reply]
The code in this repository will be safely archived for future reference.

**Reply by neyzm** [reply]
This is the most practical proposal. Having trusted active contributors takeover and maintain the project, however ensuring the community still have a say.

**Reply by CrazyWolf13** [reply]
Sounds good, I hope this community-driven idea like @MickLesk suggested it comes alive.
I like the idea with the 2/3 eye until merge principle.
So will you @MickLesk fork/create an organsation?

**Reply by MickLesk** [reply]
> Sounds good, I hope this community-driven idea like @MickLesk suggested it comes alive.
> I like the idea with the 2/3 eye until merge principle.
> So will you @MickLesk fork/create an organsation?


Yes, of course I can. But I'll wait and see (1-3 days) if there are any good ideas or similar :-)

**Reply by Mellowlynx** [reply]
I also think [MickLesk](https://github.com/MickLesk) is the one to manage the fork, as he is active.
As stated before, I will haply moderate while learning.

**Reply by Aloe-recite** [reply]
@MickLesk great idea, as long as at least 3 trusted people manually approve of new stuff after thoroughly checking it out and act as community leads and arbitrators (not sure if I got the correct translation of the word I had in mind but yo got the idea), also to avoid the future development being stolen or the community used in a malicious way or who knows what. Such great and well known communities like this one are always at risk unfortunately and there has been plenty of drama in so many projects that one must be exceptionally careful now. Kudos to you for your dedication and commitment!

**Reply by remz1337** [reply]
@MickLesk yes I'd be happy to work with you to maintain this.

**Reply by jl-678** [reply]
I have always been amazed at the scope of @tteck activities. Such amazing work!

Do we consider assigning ownership of certain scripts to specific people to simplify maintenance and upgrades? Just a thought. It would also allow more people to contribute and "specialize" in certain areas.

**Comment by tteck** [comment]
Always remember, safety first!

**Reply by Smile2Grow** [reply]
I also think that safety is one of, if not the most important topic. These scripts require real expertise. They cannot be developed using copy/paste. Even an AI cannot guarantee the aspect of secure development today or in the foreseeable future. Many people who use your scripts will probably have the need, but not the ability. The question is who can and wants to take responsibility for this? 

A second aspect for me would be the discussion of further development. I could imagine that getting into the subject and providing context are useful topics that could be conveyed on the website.

**Comment by bramsuurdje** [comment]
I'm interested in stepping up as the front-end maintainer for this project. I’ve been responsible for maintaining a redesigned website of this project I developed for a school project, which I’ve updated and maintained over time.

Github Repository: https://github.com/BramSuurdje/proxmox-helper-scripts

**Reply by deku-m** [reply]
I wonder if we can also transfer helper-scripts.com domain or make new one and make your frontend to it eventually with all scripts.
Continue the know legacy website of tteck and known URL.

Credits go to both of you.

What should we think of taking this over not only maintainers and contributers?

**Reply by tteck** [reply]
helper-scripts.com Expires: April 26, 2030 then it's up for grabs. Unless a sweat offer comes in. 😉

**Comment by Zedris** [comment]
just wanted to say @tteck. hat I had no idea of homelabbing or proxmox and have learned a ton spent many many nights learning and understanding your scripts and proxmox and overall have given me many many hours of happiness and joy. 
I truly hope all the best and love and know that you do touch peoples lives without knowing them or interacted directly with them.
<3

**Reply by kjkent** [reply]
I wonder how many lives they've touched in some way solely through contributions here -- whether it be sparking hobbies, helping to develop career skills, helping people run their businesses, entertain their families, etc. Given the popularity of the project, and the array of uses that Proxmox has in the homelab and beyond, I reckon it's a laudable number. I hope the rest of the time they have here is peaceful.

**Comment by oerix** [comment]
Instead of archiving, maybe having a few trusted contributors added to the project? Honestly speaking, I don't like the sound of forking the project, as it will just become abandonware.

**Reply by deku-m** [reply]
@tteck maybe it is possible to transfer ownership to a trustworthly...? I know this is also build in in github.

**Reply by tteck** [reply]
I was the only person I truly trusted with this code. For this reason, the repository will be archived to preserve its integrity. Anyone wishing to continue the project will need to create a fork.

**Reply by Mellowlynx** [reply]
I think he is suggesting a fork, so his original can be archived in his name.
Yes, transferring ownership is an option, but might later in however many years, erase his name over time. 
Archived, it will stay some sort of time capsule.
I get both sides, and could not make this decision.

**Reply by tteck** [reply]
I know some contributors got frustrated with my insistence on coding things a certain way, but my goal was to make the code easier to understand and consistent, rather than requiring people to learn a new approach each time.

**Reply by oerix** [reply]
> I was the only person I truly trusted with this code. For this reason, the repository will be archived to preserve its integrity. Anyone wishing to continue the project will need to create a fork.

I'm not sure how much of your time is limited, but maybe you can still manually approve new contributions when you have time? While, that being said, that will lead you to having built trust, which then you can gradually transfer ownership with your name still attached to the project.

All in all, please do what's best for you, and I'd like to thank you for the great scripts and contributions you had kindly made available for us all. 🫡

**Reply by tteck** [reply]
I have very limited time left on earth. weeks

**Reply by MickLesk** [reply]
@tteck 
It’s hard for all of us to hear what you're going through right now. No one should have to face this, and I think I’m speaking for thousands of people when I say: thank you. You’re a genius, and your scripts have made Proxmox so much more accessible for so many.

Just over a year ago, Proxmox was nearly uncharted territory for me, but through your work, I discovered its true potential. I’ve grown so much because of it—more than even in my professional life—and I’ve found so many exciting new projects along the way. Sure, there were times we didn’t always agree, but that only showed how passionate you are and how much this community means to you.

We’ll do everything we can to honor your legacy and keep reaching new people with the work you started. You’ve reached legendary status with us, and as long as Proxmox exists, your impact will never be forgotten.

**Comment by tteck** [comment]
Create a generic fork or clone e.g. `https://github.com/helper-scripts/ProxmoxVE` That gives the "helper-scripts" room to branch out if needed e.g. `https://github.com/helper-scripts/next new thing` Just a thought. It's yours to do with as you will, together.

**Reply by tteck** [reply]
I also suggest sticking with GitHub Pages for the frontend. When they say "free," they truly mean it—unlike others who lure you in with "free" only to hit you with high costs as traffic increases. Given the very high traffic this site receives, GitHub Pages is a solid choice.

**Reply by MickLesk** [reply]
I agree, GitHub Pages is the best way to ensure constant accessibility without having to rent and maintain costly servers.

For your information: 
I created a new Github and forked the project with the latest version. I have currently only adjusted the README for a short time. Is this what you had in mind? :-)

https://github.com/community-scripts/ProxmoxVE

**Reply by tteck** [reply]
I don’t have any frontend files to share, as they were never stored online. Ideally, we can find a GitHub Pages expert to set things up with a fresh community theme. Looking forward to seeing it come together!

**Reply by bramsuurdje** [reply]
I am currently working on getting my site deployed on gh-pages. should be all ready by the end of today

**Reply by remz1337** [reply]
@tteck, once we have a confirmation on the official fork (potentially looking at @MickLesk's fork), I think it would be wise to update your README before archiving to redirect people to the right place. Maybe even a notice on you website.

**Reply by tteck** [reply]
I agree

**Reply by tteck** [reply]
When the frontend is done, helper-scripts.com will point there. It now points to the GitHub Page

**Reply by bramsuurdje** [reply]
can confirm. the front-end is fully up and running. and active in prod

**Reply by tteck** [reply]
There’s still quite a bit of setup needed to get it live at https://github.com/community-scripts/ProxmoxVE. Ultimately, the community will need to decide if this design aligns with their vision, or if they’d prefer starting from scratch—this choice is up to them. 🤷🏻‍♂️

For the next steps, if the team decides to proceed, @MickLesk, @havardthom, @remz1337, @nicedevil007 can collaborate to fork the repository over. Then, they can build it to the gh-pages branch to deploy it at https://community-scripts.github.io/ProxmoxVE/, where helper-scripts will redirect.
They will also need to set up a PocketBase account to manage link corrections. For example, updating links like:
```
bash -c "$(wget -qLO - https://github.com/community-scripts/ProxmoxVE/raw/main/ct/docker.sh)"
```
Of course, this all depends on what the community decides to pursue.

**Reply by bramsuurdje** [reply]
i am willing to provide the full website. but i would still like to be the core contributor for the website. and i have some future plans like getting rid of the pocketbase database in favor of a json based system to make it easier for the community to contribute. but if need be i can update the sources of the scripts to the "new" repository immediately

**Reply by bramsuurdje** [reply]
all the scripts have been updated on my site.

**Reply by judeibe** [reply]
I like this, but wouldn’t it be beneficial to make community-scripts an organization instead of a user.

**Reply by tteck** [reply]
You may be correct, the team should look into this.

**Reply by bramsuurdje** [reply]
agree w @judeibe. way better then an account

**Reply by MickLesk** [reply]
Community-Scripts is now migrated to organisation, thx for feedback :-)

**Reply by SkilledAlpaca** [reply]
It seems that this is going to the new and dedicated fork or is there still a choice being made?

**Reply by MickLesk** [reply]
So there are all dependencies in all files updated. Now its possible to create VM / LXC from the community-project 👍🏻 
<details>
  <summary>Screenshot</summary>
  <div align="center">

  ![img](https://github.com/user-attachments/assets/6f1996e6-f56b-4893-9996-c0fa3aea2b0f)
  ![img](https://github.com/user-attachments/assets/373afec1-eb26-47d0-8871-6cc194f3734a)
  </div>
</details>

@BramSuurdje, can you update the (new) website / or is this an crawler and set the correct path automatically?
@tteck have i forgot something? I somehow remember you saying yesterday that we need to do something in Pocketbase? Or is this rather related to the website (i.e. Braam?)

**Reply by bramsuurdje** [reply]
@MickLesk all the links have been updated on the site.

**Comment by pmontgo33** [comment]
Can I ask that in whatever form this project continues, it be renamed tteck-scripts? You've touch so many during your time here, and I'm not sure thank you is enough. You've been in my thoughts since your announcement.

**Reply by tteck** [reply]
Just continuing the work is enough.

**Comment by simonmcnair** [comment]
I'm happy to host a site /moderate and/or do whatever.  I'm just not great at writing bash.  I understand it tho.  I am happy to take on paying for helper-scripts so it can remain.  Let me know.

**Comment by Outsmart2383** [comment]
tteck many thanks for your contribution. I am deeply saddened to read your latest update. I wish you the best.

It feels wrong to be discussing the future of your helper scripts, but, that is the purpose of this thread.

As someone who is an extreme novice I wanted to pose a few questions to the community:

1. What's the best way of verifying the safety of these scripts going forward? Let's pose the unlikely scenario that the "trusted members" become compromised. What steps can we take as individuals to check the code, if we can't analyse it effectively ourselves? I just pasted one of the scripts into ChatGPT and asked it to check for anything maliscous. As would be expected, it said it all looked okay, but is this a reliable? Could we create a custom GPT (or does one exist?) that can effectively scan a script for issues?
2. How are updates handled for instances that have been deployed by existing scripts? Can we continue simply updating our containers/services using the existing `update` command, or, will we need to somehow configure a link to the new repo? Is the only risk to users when a new script is used to deploy a new service, or, could a bad actor also cause existing services that were set up with the scripts to be compromised also?

**Reply by Mellowlynx** [reply]
@Outsmart2383 
1. The best way is to check a script before using it.
There are multiple ways to include malicious code/payload, and preparing for everything is not possible.
So before an update or new script is pushed, it should also be checked. (there are automated checks for this btw, but it will require a human check as well)

The installation process uses reusable scripts: _build.func_, _create_lxc.sh_, and _install.func_.
Changes to them should not happen often, and should also be extra checked, as that are used in all scripts.
But it's up to all of us to test before pushing it to main branch for example and report strange findings.
(Check the [The XZ Utils Backdoor](https://pentest-tools.com/blog/xz-utils-backdoor-cve-2024-3094) story, a sophisticated way of trying to add a backdoor in the code)

2. There will be a need for a local change, to update the script behind the `update` command to point to the new URL.
Maybe this can be done with an update to the current install scripts if they also have the update function in them, or a separate `wget` command with a scrip that makes the changes.
Another option is a one-liner that will update everything for you.

This is something that needs to be checked and then made and deployed.

**Comment by wisie** [comment]
Just want to say thanks for everything you've done. I check the change log far too often then I should admit as I'm always interested in what's recently been packaged. Wishing you all the best and thank you 🙏

**Comment by JamesMowery** [comment]
Just want to let you know that you rock @tteck 

You helped make the concept of homelab (for someone like me who had never done anything like that) a reality with your efforts. Just know that my appreciation (and hopefully from many others) is eternal!

**Comment by Po3Tato** [comment]
@tteck Thanks for the amazing scripts. Saved me a lot of times and can't thank you enough. Thanks for your efforts.

**Comment by havardthom** [comment]
Hi all, great to see the community coming together to carry on this fantastic project. I have not been a contributor for very long, but I am happy to join in maintaining a community effort if you'd like, though I can not promise to be as active as tteck/others have been, this is mostly an evening/weekend hobby for me.

Reading through this thread I have some initial thoughts for discussion
- I agree that a GitHub organization is the way to go for a community project
- Main branch protection is probably a good idea since this is not a one man show anymore
- PRs requiring 2 approvals - it will slow down development but worth it for security aspect
- For a frontend I personally wouldn't start a next.js project, mostly because I agree with the sentiments in this article https://www.epicweb.dev/why-i-wont-use-nextjs. This is all subjective though and if the goal is getting up and running as quick as possible perhaps an existing frontend is the best choice. Alternatively we can start from scratch for full control.
- We are only serving static content, why do we need a database? Who is gonna host it, who is gonna manage it, who is gonna pay for it? Only complicates things imo.
    - For docs, perhaps we only need a single json file per script, and since it is tightly coupled with scripts, perhaps transparently stored in a folder similar to `ct` and `install` following conventions. Since the scripts are so standardized, some parts could even be generated in the future, allowing changes in script specs without manually updating docs. Just an idea.

**Reply by bramsuurdje** [reply]
Just a few words from me regarding the front-end.

As for your 4th point, I do personally use NextJS for all my projects, and since it’s the most popular framework, I feel like it’s best since most people will theoretically know how to contribute.

The actual article is all personal experience, and he even says, "Whatever you use is probably fine.
Your tool choice matters much less than your skill at using the tool to accomplish your desired outcome (a great user experience)."

As for the database, I have started work on a JSON-based system, so it’s easier for the community to also contribute to it. This will remove the need for a database. At first a database was nice because it provided a GUI for me and others to add scripts. But now that it’s not a necessity anymore, I’m planning on replacing it for a more intuitive system. Keep an eye out for a new branch being made for developing that. 

As for who is paying for the database, that is currently me. Although the database is getting no traffic whatsoever, so it doesn’t cost me anything.

**Comment by wiebereu** [comment]
How can we make sure that the helper scripts are secure and that some people don't abuse this situation to breach other people's security when it comes to Proxmox? I hope we don't end up with multiple forks and that we understand that scripts need to be safe. That is the reason many people trust the helper scripts from Tteck. I hope the trust stays

**Reply by MickLesk** [reply]
Look at discussion above from @Outsmart2383 and @Mellowlynx 👍
